service/task_manager_module.cc

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "locator/tablets.hh"
+#include "replica/database.hh"
+#include "service/storage_service.hh"
+#include "service/task_manager_module.hh"
+#include "tasks/task_handler.hh"
+#include "tasks/virtual_task_hint.hh"
+
+namespace service {
+
+tasks::task_manager::task_group tablet_virtual_task::get_group() const noexcept {
+    return tasks::task_manager::task_group::tablets_group;
+}
+
+future<std::optional<tasks::virtual_task_hint>> tablet_virtual_task::contains(tasks::task_id task_id) const {
+    auto tables = get_table_ids();
+    for (auto table : tables) {
+        auto& tmap = _ss.get_token_metadata().tablets().get_tablet_map(table);
+        for (const locator::tablet_info& info : tmap.tablets()) {
+            if (info.repair_task_info.is_valid() && info.repair_task_info.tablet_task_id.uuid() == task_id.uuid()) {
+                co_return tasks::virtual_task_hint{
+                    .table_id = table,
+                };
+            }
+        }
+    }
+    co_return std::nullopt;
+}
+
+future<tasks::is_abortable> tablet_virtual_task::is_abortable(tasks::virtual_task_hint hint) const {
```

Comment:
`hint` param is unused, will it be used in a later patch?

Reply:
In this PR - no. It will be used later though. I can remove it for now, but I think it makes sense to pass the hint to all public virtual task methods

Reply:
Ok, it can stay if you have plans for it.
When this is the case, best to mention this in the commitlog, so reviewers are not left guessing if this is a mistake or part of a plan.